src/shared/extension.ts

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+export {}
+
+// Causing colorization issues if placed above Array.prototype...
+// Ideally: ((_) => number) | ((_) => string)
+type Selector<T> = (_: T) => number | string
+
+declare global {
+	interface ArrayConstructor {
+		commonLength(a: any[], b: any[]): number
+	}
+	interface Array<T> {
+		last: T
+		replace(items: T[]): void // From Mobx, but not showing up.
+		remove(item: T): boolean // From Mobx, but not showing up.
+		removeWhere(predicate: (item: T) => boolean): T | false
+		sortBy<T>(this: T[], selector: Selector<T>, descending?: boolean): Array<T> // Not a copy
+	}
+	interface String {
+		file: string
+		path: string
```

Comment:
This bothers me a bit since we should probably be treating everything as URIs, so I could see these extensions on VSCode's URI class.

I know there are places were we have to treat them as strings, but those should be limited to the UI.

If these are only for UI display, then calling them something like fileDisplayName/pathDisplayName would make it clear these strings shouldn't be used for URI manipulation.

Reply:
The issue is that we don't have access to the Uri class from within the WebView. Yes, ideally it would be extended from the Uri class. Any ideas for workarounds?

Reply:
The issue is that we don't have access to the Uri class from within the WebView. Yes, ideally it would be extended from the Uri class. Any ideas for workarounds?

---
In reply to: [435600491](https://github.com/microsoft/sarif-vscode-extension/pull/279#discussion_r435600491) [](ancestors = 435600491)